spring-batch-infrastructure/src/main/java/org/springframework/batch/support/transaction/TransactionAwareBufferedWriter.java

```diff
@@ -113,18 +162,11 @@ public void beforeCommit(boolean readOnly) {
 				}
 
 				private void complete() throws IOException {
-					StringBuilder buffer = (StringBuilder) TransactionSynchronizationManager.getResource(bufferKey);
+					Buffer buffer = (Buffer) TransactionSynchronizationManager.getResource(bufferKey);
 					if (buffer != null) {
-						String string = buffer.toString();
-						byte[] bytes = string.getBytes(encoding);
-						int bufferLength = bytes.length;
-						ByteBuffer bb = ByteBuffer.wrap(bytes);
-						int bytesWritten = channel.write(bb);
-						if(bytesWritten != bufferLength) {
-							throw new IOException("All bytes to be written were not successfully written");
```

Comment:
This can no longer happen because we now call `java.io.ByteArrayOutputStream#writeTo(OutputStream)` which writes the entire buffer (and propagates any exception).